server/src/main/java/org/opensearch/common/metrics/MetricsTracker.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.metrics;
+
+import com.sun.management.ThreadMXBean;
+import org.opensearch.common.SuppressForbidden;
+
+import java.lang.management.ManagementFactory;
+
+/*
+    Simple tracker for CPU consumption and memory allocated by the current thread.
+    In the current model, it is important for updateMetrics to be called from the thread that created the object
+
+    This will evolve to support frequent updates and directly update task with resource usage
+*/
+@SuppressForbidden(reason = "ThreadMXBean enables tracking resource consumption by a thread. "
+    + "It is platform dependent and i am not aware of an alternate mechanism to extract this info")
+public class MetricsTracker {
+
+    static ThreadMXBean threadMXBean = (ThreadMXBean) ManagementFactory.getThreadMXBean();
+    long startingAllocatedBytes;
+    long startingCPUTime;
```

Comment:
It looks like these values are never assigned. Did you intend to initialize these to the initial measurements upon instance creation?

Reply:
thanks for catching this. The constructor got removed after the recent refactor/reset removal! fixing it.